core/src/test/scala/unit/kafka/admin/ListConsumerGroupTest.scala

```diff
@@ -111,37 +233,138 @@ class ListConsumerGroupTest extends ConsumerGroupCommandTest {
+  }
+
+  @ParameterizedTest(name = TestInfoUtils.TestWithParameterizedQuorumAndGroupProtocolNames)
+  @MethodSource(Array("getTestQuorumAndGroupProtocolParametersConsumerGroupProtocolOnly"))
+  def testListGroupCommandConsumerProtocol(quorum: String, groupProtocol: String): Unit = {
+    val simpleGroup = "simple-group"
+    val protocolGroup = "protocol-group"
+
+    createOffsetsTopic()
+
+    addSimpleGroupExecutor(group = simpleGroup)
+    addConsumerGroupExecutor(numConsumers = 1)
+    addConsumerGroupExecutor(numConsumers = 1, group = protocolGroup, groupProtocol = groupProtocol)
+    var out = ""
+
+    var cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      !out.contains("STATE") && out.contains(simpleGroup) && out.contains(group) && out.contains(protocolGroup)
+    }, s"Expected to find $simpleGroup, $group, $protocolGroup and no header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--state")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("STATE") && !out.contains("TYPE") && out.contains(simpleGroup) && out.contains(group) && out.contains(protocolGroup)
+    }, s"Expected to find $simpleGroup, $group, $protocolGroup and the header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--type")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && out.contains("Consumer") && !out.contains("STATE") && out.contains(protocolGroup) && out.contains(simpleGroup) && out.contains(group)
+    }, s"Expected to find $protocolGroup and the header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--type", "consumer")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && out.contains("Consumer") && !out.contains("STATE") && out.contains(protocolGroup)
+    }, s"Expected to find $protocolGroup and the header, but found $out")
+
+    cgcArgs = Array("--bootstrap-server", bootstrapServers(), "--list", "--type", "consumer", "--state", "Stable")
+    TestUtils.waitUntilTrue(() => {
+      out = TestUtils.grabConsoleOutput(ConsumerGroupCommand.main(cgcArgs))
+      out.contains("TYPE") && out.contains("Consumer") && out.contains("STATE") && out.contains("Stable") && out.contains(protocolGroup)
+    }, s"Expected to find $protocolGroup and the header, but found $out")
   }
 
+  def testWaitUntilTrue(
+    typeFilterSet: Set[GroupType],
+    stateFilterSet: Set[ConsumerGroupState],
+    expectedListing: Set[ConsumerGroupListing],
+    service:  ConsumerGroupCommand.ConsumerGroupService
```

Comment:
nit: There is an extra space after `:`. I also suggest to put `service` first.

Reply:
ugh idky checkstyle didn't catch this when I ran locally :( thanks